quickstart/quickstart.Dockerfile

```diff
@@ -0,0 +1,87 @@
+# ---------------------------------------------
+# Docker Stage: Base builder used for both parsec service and tools
+FROM rust:latest AS base_builder
+
+RUN apt update -y && \
+    apt install -y llvm-dev libclang-dev clang cmake jq
+
+## Track various build environment things we may want to use throughout
+WORKDIR /build-env
+RUN echo "$(uname | awk '{print tolower($0)}')" > /build-env/os
+RUN echo "$(arch)" > /build-env/arch
+
+# ---------------------------------------------
+# Docker Stage: Temporary stage to help dependency caching
+FROM base_builder AS parsec_service_scratch
+
+## Copy everything in
+COPY . /parsec-service
+WORKDIR /parsec-service
+
+# This just adds cargo dependencies to the scratch stage so that we don't need to
+# download them each time the builder runs.
+RUN cargo fetch
+
+# Save the current parsec version as defined by cargo
+RUN echo "$(cargo metadata --format-version=1 --no-deps --offline | jq -r '.packages[0].version')" > /build-env/parsec-version
+
+# ---------------------------------------------
+# Docker Stage: Executes the build of the Parsec Service
+FROM parsec_service_scratch AS parsec_service_builder
+
+## Run the actual build
+RUN cargo build --release --features mbed-crypto-provider
+
+# ---------------------------------------------
+# Docker Stage: Executes the build of the Parsec Tool
+FROM base_builder AS parsec_tool_builder
+
+RUN git clone https://github.com/parallaxsecond/parsec-tool /parsec-tool
+WORKDIR /parsec-tool
+RUN git checkout $(git tag --sort=committerdate | tail -1)
+RUN cargo build --release
+
+# ---------------------------------------------
+# Docker Stage: Extracts build results from previous stages and adds in quickstart configs
+FROM base_builder AS layout
+
+## Add the built binaries into the image and make available on PATH
```

Comment:
I need to remove ` and make available on PATH`. That step occurs in a later stage.